implementations/rust/ockam/ockam/src/entity/issuer.rs

```diff
@@ -0,0 +1,123 @@
+use crate::{
+    Address, CredentialAttribute, CredentialFragment2, CredentialIssuer, CredentialOffer,
+    CredentialRequest, CredentialSchema, OckamError, OfferIdBytes, ProfileTrait, ProofBytes,
+    PublicKeyBytes, Result, Route,
+};
+
+pub struct LocalIssuingEntity<E: ProfileTrait> {
+    address: Address,
+    credential_issuer: CredentialIssuer,
+    credential_schema: CredentialSchema,
+    _entity: E,
+    public_key: PublicKeyBytes,
```

Comment:
How is PublicKeyBytes different from PublicKey?

Reply:
`PublicKeyBytes` is a type for the raw slice key material. I added it a few months ago to begin cleaning up bare usages of arrays in return signatures like `[u8 : 96]`. It looks like `PublicKey` is similar but a heapless vec instead of a slice. The differences are originating from the credentials API, so we may need to work `PublicKey` into there more.